libbeat/docs/security/api-keys.asciidoc

```diff
@@ -0,0 +1,116 @@
+[role="xpack"]
+[[beats-api-keys]]
+=== Grant access using API keys
+
+Instead of using usernames and passwords you can use API keys to grant
+access to {es} resources. API keys can be set to expire at a certain time
+and explicitly invalidated. Any user with the `manage_api_key` or `manage_own_api_key`
+cluster privilege can create API keys.
+
+See the {es} API key documentation for more information:
+
+* {ref}/security-api-create-api-key.html[Create API key]
+* {ref}/security-api-get-api-key.html[Get API key information]
+* {ref}/security-api-invalidate-api-key.html[Invalidate API key]
+
+Every {beatname_uc} instance is typically sending both collected data and monitoring
+information to {es}. If you are sending both to the same cluster, you can use the same
+API key. For different clusters, you will need an API key per cluster.
+
+NOTE: For security reasons, we recommend using a unique API key per Beat instance.
+You can create as many API keys per user as necessary.
+
+[[beats-api-key-publish]]
+==== Create an API key for publishing
+To create an API key to use for writing data to {es}, use the
+{ref}/security-api-create-api-key.html[Create API key API], for example:
+
+[source,console,subs="attributes,callouts"]
+------------------------------------------------------------
+POST /_security/api_key
+{
+  "name": "{beat_default_index_prefix}_host001", <1>
+  "role_descriptors": {
+    "{beat_default_index_prefix}_writer": { <2>
+      "cluster": ["monitor", "read_ilm"],
+      "index": [
+        {
+          "names": ["{beat_default_index_prefix}-*"],
+          "privileges": ["view_index_metadata", "create_doc"]
+        }
+      ]
+    }
+  }
+}
+------------------------------------------------------------
+<1> name of the API key
```

Comment:
I don't feel strongly here, but most editors would tell you to use the same style here (capitalization, etc) that you use for other callouts. Minor point IMO.